{
    "dependencies": [
        {
            "name": "pyyaml",
            "version": "5.1.2",
            "vulns": [
                {
                    "id": "PYSEC-2020-176",
                    "fix_versions": [
                        "5.2b1"
                    ],
                    "aliases": [
                        "GHSA-3pqx-4fqf-j49f",
                        "CVE-2019-20477"
                    ],
                    "description": "PyYAML 5.1 through 5.1.2 has insufficient restrictions on the load and load_all functions because of a class deserialization issue, e.g., Popen is a class in the subprocess module. NOTE: this issue exists because of an incomplete fix for CVE-2017-18342."
                },
                {
                    "id": "PYSEC-2020-96",
                    "fix_versions": [
                        "5.3.1"
                    ],
                    "aliases": [
                        "CVE-2020-1747",
                        "GHSA-6757-jp84-gxfx"
                    ],
                    "description": "A vulnerability was discovered in the PyYAML library in versions before 5.3.1, where it is susceptible to arbitrary code execution when it processes untrusted YAML files through the full_load method or with the FullLoader loader. Applications that use the library to process untrusted input may be vulnerable to this flaw. An attacker could use this flaw to execute arbitrary code on the system by abusing the python/object/new constructor."
                },
                {
                    "id": "PYSEC-2021-142",
                    "fix_versions": [
                        "5.4"
                    ],
                    "aliases": [
                        "GHSA-8q59-q68h-6hv4",
                        "CVE-2020-14343"
                    ],
                    "description": "A vulnerability was discovered in the PyYAML library in versions before 5.4, where it is susceptible to arbitrary code execution when it processes untrusted YAML files through the full_load method or with the FullLoader loader. Applications that use the library to process untrusted input may be vulnerable to this flaw. This flaw allows an attacker to execute arbitrary code on the system by abusing the python/object/new constructor. This flaw is due to an incomplete fix for CVE-2020-1747."
                }
            ]
        },
        {
            "name": "requests",
            "version": "2.32.3",
            "vulns": []
        },
        {
            "name": "certifi",
            "version": "2025.1.31",
            "vulns": []
        },
        {
            "name": "charset-normalizer",
            "version": "3.4.1",
            "vulns": []
        },
        {
            "name": "idna",
            "version": "3.10",
            "vulns": []
        },
        {
            "name": "urllib3",
            "version": "2.3.0",
            "vulns": []
        }
    ],
    "fixes": [
        {
            "name": "pyyaml",
            "old_version": "5.1.2",
            "new_version": "5.4"
        }
    ]
}
